domain/domain.go

```diff
@@ -259,18 +259,18 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 }
 
 // Returns the timestamp of a schema version, which is the commit timestamp of the schema diff
-func (do *Domain) getTimestampForSchemaVersionWithNonEmptyDiff(m *meta.Meta, version int64) (int64, error) {
+func (do *Domain) getTimestampForSchemaVersionWithNonEmptyDiff(m *meta.Meta, version int64, startTS uint64) (int64, error) {
 	tikvStore, ok := do.Store().(helper.Storage)
 	if ok {
-		helper := helper.NewHelper(tikvStore)
-		data, err := helper.GetMvccByEncodedKey(m.EncodeSchemaDiffKey(version))
+		newHelper := helper.NewHelper(tikvStore)
+		mvccResp, err := newHelper.GetMvccByEncodedKeyWithTS(m.EncodeSchemaDiffKey(version), startTS)
```

Comment:
Can we just call another `GetMvccByEncodedKey(lock.primary)` to get the commit-ts? Resolving lock and retring makes GetMvcc complex and heavy.

Reply:
It's possible the primary key is not committed or the ongoing internal transaction is invisible to the input `start_ts`.

Reply:
Besides, it's better to use snapshot interfaces instead of the mvcc interfaces, I've filed an issue about it https://github.com/pingcap/tidb/issues/48283